Sources/MembraneRTC/Media/SoundDetection.swift

```diff
@@ -0,0 +1,179 @@
+import AVFoundation
+
+// Class for sound detection using the device's microphone
+public class SoundDetection: ObservableObject {
+
+  // Audio engine for sound detection
+  private var engine: AVAudioEngine!
+  private var inputNode: AVAudioInputNode!
+  private var mixerNode: AVAudioMixerNode!
+  private var onSoundDetectedListener: ((_ detectionResult: Bool) -> Void)?
+  private var onVolumeChangedListener: ((_ volume: Int) -> Void)?
+
+  // Property to indicate the recording state
+  @Published public var isRecording: Bool = false
+
+  /// Sets a listener to receive sound volume change events.
+  ///
+  /// - Parameter listener: The listener to be notified when sound volume changes.
```

Comment:
if you format it nicely like this:
```
    /**
        Updates maximum bandwidth for the given simulcast encoding of the given track.

        - Parameters:
         - trackId: track id of a video track
         - encoding: rid of the encoding
         - bandwidth: bandwidth in kbps
     */
```
xcode will display it nicely and the docs should display it nicely